settings.php

```diff
@@ -26,14 +26,136 @@
+        'ap-southeast-2' => 'Asia Pacific (Sydney)',
+        'eu-west-1'      => 'EU (Ireland)',
+    );
+    $settings->add(new admin_setting_configselect('local_smartmedia/api_region',
+        get_string('settings:aws:region', 'local_smartmedia'),
+        get_string('settings:aws:region_help', 'local_smartmedia'),
+        'ap-southeast-2',
+        $regionoptions));
+
+    $settings->add(new admin_setting_configtext('local_smartmedia/sqs_queue_url',
+        get_string('settings:aws:sqs_queue_url', 'local_smartmedia'),
+        get_string('settings:aws:sqs_queue_url_help', 'local_smartmedia'),
+        '',
+        PARAM_URL));
+
+    // Output settings.
+    $settings->add(new admin_setting_heading('outputheader',
+        get_string('settings:output:header', 'local_smartmedia'),
+        get_string('settings:output:header_desc', 'local_smartmedia')));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/quality_low',
+        get_string('settings:output:quality_low', 'local_smartmedia'),
+        get_string('settings:output:quality_low_help', 'local_smartmedia'), 0));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/quality_medium',
+        get_string('settings:output:quality_medium', 'local_smartmedia'),
+        get_string('settings:output:quality_medium_help', 'local_smartmedia'), 0));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/quality_high',
+        get_string('settings:output:quality_high', 'local_smartmedia'),
+        get_string('settings:output:quality_high_help', 'local_smartmedia'), 0));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/audio_output',
+        get_string('settings:output:audio_output', 'local_smartmedia'),
+        get_string('settings:output:audio_output_help', 'local_smartmedia'), 0));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/download_files',
+        get_string('settings:output:download_files', 'local_smartmedia'),
+        get_string('settings:output:download_files_help', 'local_smartmedia'), 0));
+
+    // Processing settings.
+    $settings->add(new admin_setting_heading('processingheader',
+        get_string('settings:processing:header', 'local_smartmedia'),
+        get_string('settings:processing:header_desc', 'local_smartmedia')));
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/proactiveconversion',
+        get_string('settings:processing:proactiveconversion', 'local_smartmedia'),
+        get_string('settings:processing:proactiveconversion_help', 'local_smartmedia'), 1));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/viewconversion',
+        get_string('settings:processing:proactiveconversion', 'local_smartmedia'),
+        get_string('settings:processing:proactiveconversion_help', 'local_smartmedia'), 1));
+
+    $settings->add(new admin_setting_configduration('local_smartmedia/convertfrom',
```

Comment:
This must be a date not a duration, I don't want to change the semantics of any settings here, just the UI

Reply:
hmm, there is no amdin setting which is a date picker, is that why you did it this way?

Reply:
yes, I couldn't find a similar element.

Reply:
I also think the lang strings for this element should be touched to reflect that its a duration backwards, as opposed to a date forwards

Reply:
Also as we are changing the semantics of this we should have an upgrade script that converts from a timestamp to a duration.

It's gonna be a bit weird because they absolute date might be say 105 days, 4 hours, 10 seconds ago, so I think find the time delta and then round it to the nearest whole day and the save it again.

Reply:
This has been added, and the default for the control set to 1 week.